test/modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -87,6 +64,41 @@ abstract class DirectMethodSenderBase : IDisposable
             }
         }
 
+        async Task<int> InvokeDirectMethodWithRetryAsync(
+            ILogger logger,
+            string deviceId,
+            string targetModuleId,
+            string methodName,
+            ulong directMethodCount)
+        {
+            const int maxRetry = 3;
+            int resultStatus = (int)HttpStatusCode.InternalServerError;
+            int transientRetryCount = 0;
+
+            while (transientRetryCount <= maxRetry)
```

Comment:
If this if-condition has equal sign, it will run maxRetry+1 times.  Therefore remove equal sign from if-condition.